service-map/src/main/java/fi/nls/oskari/wmts/domain/TileMatrixLimits.java

```diff
@@ -1,57 +1,74 @@
 package fi.nls.oskari.wmts.domain;
 
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+
 /**
- * Created with IntelliJ IDEA.
- * User: SMAKINEN
- * Date: 6.6.2014
- * Time: 14:15
- * To change this template use File | Settings | File Templates.
+ * @see http://schemas.opengis.net/wmts/1.0/wmtsGetCapabilities_response.xsd
  */
 public class TileMatrixLimits {
 
-    private String tileMatrix;
-    private int minTileRow;
-    private int maxTileRow;
-    private int minTileCol;
-    private int maxTileCol;
+    private static Logger LOG = LogFactory.getLogger(TileMatrixLimits.class);
+
+    private final TileMatrix tm;
+    private final int minTileRow;
+    private final int maxTileRow;
+    private final int minTileCol;
+    private final int maxTileCol;
 
-    public String getTileMatrix() {
-        return tileMatrix;
+    public TileMatrixLimits(TileMatrix tm, int minTileRow, int maxTileRow,
+            int minTileCol, int maxTileCol) {
+        this.tm = tm;
+        this.minTileRow = minTileRow;
+        this.maxTileRow = maxTileRow;
+        this.minTileCol = minTileCol;
+        this.maxTileCol = maxTileCol;
+        validate();
     }
 
-    public void setTileMatrix(String tileMatrix) {
-        this.tileMatrix = tileMatrix;
+    /**
+     * Just log as warnings instead of throwing exceptions for now
+     * TODO: Don't accept invalid values
+     */
+    private void validate() {
+        if (minTileRow < 0) {
+            LOG.warn(tm.getId(), "Negative MinTileRow");
+        }
+        if (maxTileRow >= tm.getMatrixWidth()) {
```

Comment:
First thought was that rows should be compared to height, but looking at the schema this looks correct. Maybe add a comment about this since it's a bit confusing:


    <element name="MaxTileCol" type="positiveInteger">
          <annotation>
                <documentation>
                         Maximim tile column index valid for this layer. From minTileCol to tileHeight-1 of the tileMatrix section of this tileMatrixSet.


      <element name="MaxTileRow" type="positiveInteger">
                <annotation>
                       <documentation>
                               Maximim tile row index valid for this layer. From minTileRow to matrixWidth-1 of the tileMatrix section of this tileMatrixSet